File: ios/Classes/Features/AgentShutdown.swift
import ADEUMInstrumentation
import Flutter

extension SwiftAppDynamicsAgentPlugin {
  func shutdownAgent(result: @escaping FlutterResult, arguments: Any?) {
    ADEumInstrumentation.shutdownAgent()
    result(nil)
  }
  
  func restartAgent(result: @escaping FlutterResult, arguments: Any?) {
    ADEumInstrumentation.restartAgent()
    result(nil)
  }
}

File: android/src/main/kotlin/com/appdynamics/appdynamics_mobilesdk/features/AgentShutdown.kt
/*
 * Copyright (c) 2021. AppDynamics LLC and its affiliates.
 * All rights reserved.
 *
 */

package com.appdynamics.appdynamics_mobilesdk.features

import androidx.annotation.NonNull
import com.appdynamics.appdynamics_mobilesdk.AppDynamicsMobileSdkPlugin
import com.appdynamics.eumagent.runtime.Instrumentation
import io.flutter.plugin.common.MethodChannel

fun AppDynamicsMobileSdkPlugin.shutdownAgent(
    @NonNull result: MethodChannel.Result,
    arguments: Any?
) {
    Instrumentation.shutdownAgent();
    result.success(null)
}

fun AppDynamicsMobileSdkPlugin.restartAgent(
    @NonNull result: MethodChannel.Result,
    arguments: Any?
) {
    Instrumentation.restartAgent();
    result.success(null)
}
